lesson16/index.js

```diff
@@ -101,47 +113,86 @@ function createCommentInfo(values) {
     return fragment;
 }
 
-async function createArticleTitle(data) {
+async function createArticleContents(data) {
     const values = data.map((value) => value.articles);
     const tabContents = document.getElementById("js-tabContents");
     const tabContentsInner = document.getElementById("js-tabContentsInner");
 
     //記事データの数だけulを作成
     for (let i = 0; i < values.length; i++) {
         const ul = document.createElement("ul");
-        ul.id = `js-tabContentsList${i+1}`;
+        ul.id = `js-${data[i].category}-contents`;
         ul.classList.add("tab__contents-list", "js-tabContentsList");
 
         const articleTitlesFragment = appendArticlesTitleFragment(values[i]);
+        const contentsImgFragment = createImgFragments(data[i]);
 
         tabContents.appendChild(tabContentsInner).appendChild(ul).appendChild(articleTitlesFragment);
+        tabContentsInner.appendChild(contentsImgFragment);
     }
+}
 
-    // TODO 今回のPRで「どのカテゴリタブを初期表示時に選んでいるかはデータとして持っている」を実装していないため仮に作ってある。後で実装する
-    const tabContentsItem = document.getElementById("js-tabContentsList1");
-    tabContentsItem.classList.add("is-show");
+function createImgFragments(data) {
+    const fragment = document.createDocumentFragment();
+    const imgWrapper = document.createElement("div");
+    const img = document.createElement("img");
+
+    imgWrapper.classList.add("tab__img-wrapper", "js-contentsImgWrapper");
+    imgWrapper.id = `js-${data.category}-imgWrapper`
+    img.classList.add("tab__img");
+    img.src = `${data.img}`;
+
+    fragment.appendChild(imgWrapper).appendChild(img);
+
+    return fragment;
+}
+
+async function fetchInitDisplayData(values) {
+    const initDisplayData = values.find(value => value.display === true);
+    const initDisplayCategory = initDisplayData.category;
+    const navButton = document.getElementById(initDisplayCategory);
+    const tabContentsItem = document.getElementById(`js-${initDisplayCategory}-contents`);
+    const tabContentsImg = document.getElementById(`js-${initDisplayCategory}-imgWrapper`);
+
+    navButton.classList.add("is-active");
+    tabContentsItem.classList.add("is-active");
+    tabContentsImg.classList.add("is-active");
 }
 
 async function addTabContents() {
-    const data = await fetchData();
+    const data = await fetchArrayData();
 
     createTabNav(data);
     createTabContainer();
     createTabContents();
```

Comment:
IMO
If an error occurs, undefined will be passed to data.
In this case, it would be better to make a conditional branch so that it will not be processed.
```suggestion
    const data = await fetchArrayData();
     if(data) {
    createTabNav(data);
    createTabContainer();
    createTabContents();
     }

```

Reply:
I hadn't thought about it at all, so I learned a lot.
I fixed it. Thank you!
[e867995](https://github.com/haru-programming/JavaScript-lessons/pull/19/commits/e867995a9d4d02609936f3f5f97232f3043e8404)